tests/form_params_test.go

```diff
@@ -0,0 +1,269 @@
+// duplicated and of equal
+// length, the request will succeed, otherwise it fails. This conforms to RFC
+// 7320, Section 3.3.2 which says that the server can either replace duplicate
+// Content-Length headers with one value or reject the request. However, for a
+// list of identical values the request is rejected, which is rather inconsistent.
+func TestFormParametersDuplicateContentLength(t *testing.T) {
+	t.Skip()
+	reqBody := "veggie=potato"
+	postReq := []byte("POST / HTTP/1.1\r\n" + "Host: localhost:8080\r\n" + "Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" + "Content-Length: 13\r\n" + "Content-Length: 13\r\n" + "\r\n" + reqBody + "\r\n" + "\r\n")
+	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		if contentLen := req.Header["Content-Length"][0]; contentLen != "13" {
+			t.Errorf("Expected Content-Length to be 13 but got %s", contentLen)
+			if val := req.FormValue("veggie"); val != "potato" {
+				t.Errorf("req.FormValue: want potato, got %s", val)
+			}
+		}
+	})
+	if err != nil {
+		t.Errorf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !isOkReq(resp) {
+		t.Errorf("response: want %s, got %s", statusOK, string(resp))
+	}
+
+	reqBody = "veggie=potato"
+	postReq = []byte("POST / HTTP/1.1\r\n" + "Host: localhost:8080\r\n" + "Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" + "Content-Length: 13, 13\r\n" + "\r\n" + reqBody + "\r\n" + "\r\n")
+	resp, err = requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		if contentLen := req.Header["Content-Length"][0]; contentLen != "13" {
+			t.Errorf("Expected Content-Length to be 13 but got %s", contentLen)
+			if val := req.FormValue("veggie"); val != "potato" {
+				t.Errorf("req.FormValue: want potato, got %s", val)
+			}
+		}
+	})
+	if err != nil {
+		t.Errorf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !isOkReq(resp) {
+		t.Errorf("response: want %s, got %s", statusOK, string(resp))
+	}
+
+	postReq = []byte("POST / HTTP/1.1\r\n" + "Host: localhost:8080\r\n" + "Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" + "Content-Length: 13\r\n" + "Content-Length: 12\r\n" + "\r\n" + reqBody + "\r\n" + "\r\n")
+	resp, err = requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+		if contentLen := req.Header["Content-Length"][0]; contentLen != "13" {
+			t.Error("MakeRequest(): request should have been rejected when list of values is given to Content-Length, rather inconsistent if you ask me.")
+		}
+		req.ParseForm()
+
+	})
+	if err != nil {
+		t.Errorf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !isBadReq(resp) {
+		t.Errorf("response: want %s, got %s", statusBadReq, string(resp))
+	}
+}
+
+// Test whether form parameters with Content-Type:
+// application/x-www-form-urlencoded that break percent-encoding will be
+// ignored
+func TestFormParametersBreakUrlEncoding(t *testing.T) {
+
```

Comment:
nit: Remove extra newline in start of function

Reply:
Resolved in 12a32a